### PR TITLE
ES6: meta property `new.target`.

### LINKED ISSUE #1203: ES6 feature: Meta property
## Syntax:

<pre>
NewTarget : new . target
</pre>

## Spec:

https://people.mozilla.org/~jorendorff/es6-draft.html#sec-meta-properties

Relevant AST interfaces (from https://github.com/estree/estree):

```js
interface MetaProperty <: Expression {
    type: "MetaProperty";
    meta: Identifier;
    property: Identifier;
}
```

### PR DESCRIPTION
**Pull request by ariya**
Closes #1203.

### CODE REVIEW

File: test/fixtures/ES6/meta-property/assign-new-target.tree.json [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+                                            "column": 9
+                                        }
+                                    },
+                                    "type": "Identifier",
+                                    "name": "x"
+                                },
+                                "init": {
+                                    "range": [
+                                        27,
+                                        0
```
Comment (ikarienator) [inline test/fixtures/ES6/meta-property/assign-new-target.tree.json]: The range is incorrect.

**Comment by ikarienator** [comment]
0. I think we should add more tests to make sure the expression precedency is correct.
0. `NewTarget` is not allowed in anywhere out side of a function-like. We should check it.

Otherwise LGTM :+1:.

**Comment by ariya** [comment]
For the operator precedence test, what will be a suitable test case?


 --
Ariya Hidayat, twitter.com/AriyaHidayat

File: test/fixtures/ES6/meta-property/assign-new-target.tree.json (hunk 1 above)
Comment (ariya) [inline test/fixtures/ES6/meta-property/assign-new-target.tree.json]: Oops, good catch, thank you!

**Comment by ikarienator** [comment]
For example, in `new new.target()()`, the first `new` should be combined with the first pair of parentheses but not the second. The result should be a call expression but not a new expression. This can potentially be interfering with the nature of LL(1) parser. We are parsing this case correctly though. Not a problem.

**Comment by ariya** [comment]
I've added the precedence test. Anything else we shall consider before I'm landing this one?

**Pull request closed without merging**